pkg/cgroups/cgroups_test.go

```diff
@@ -4,26 +4,57 @@
 package cgroups
 
 import (
+	"fmt"
+	"os"
 	"path/filepath"
 	"syscall"
 	"testing"
 
+	"github.com/cilium/tetragon/pkg/defaults"
+	"github.com/cilium/tetragon/pkg/mountinfo"
 	"github.com/stretchr/testify/assert"
 	"golang.org/x/sys/unix"
 )
 
-func TestDetectCgoupModeInvalid(t *testing.T) {
+func isDirMountFsType(path string, mntType string) (bool, error) {
+	st, err := os.Stat(path)
+	if err != nil {
+		return false, nil
+	}
+
+	if !st.IsDir() {
+		return false, fmt.Errorf("%s is not a directory", path)
+	}
+
+	infos, err := mountinfo.GetMountInfo()
+	if err != nil {
+		return false, err
+	}
+
+	mounted, instance := mountinfo.IsMountFS(infos, mntType, path)
+	if !mounted {
+		return false, fmt.Errorf("%s directory exists but not mounted as %s! left-over?", path, mntType)
+	}
+
+	if !instance {
+		return false, fmt.Errorf("%s has different mount system than %s", path, mntType)
+	}
+
+	return true, nil
+}
+
+func TestDetectCgroupModeInvalid(t *testing.T) {
 	mode, err := detectCgroupMode("invalid-cgroupfs-path")
 	assert.Error(t, err)
 	assert.Equal(t, CGROUP_UNDEF, mode)
 }
 
-func TestDetectCgoupModeSimple(t *testing.T) {
+func TestDetectCgroupModeDefault(t *testing.T) {
```

Comment:
Wasn't this added in the previous commit? If so, please edit the previous commit rather than adding the change in this commit; unless there's a good reason of course!

Reply:
Indeed, I fixed this and added more comments to explain what the tests do including commit logs, much appreciated @kevsecurity